dashboard/app/controllers/regional_partners_controller.rb

```diff
@@ -109,6 +114,35 @@ def remove_mapping
     redirect_to @regional_partner
   end
 
+  # POST /regional_partners/:id/replace_mappings
+  # Replace mappings for partner with given id with contents
+  # of CSV file passed in regions parameter
+  # CSV file must have header 'Region' and all regions must be
+  # valid (zip code or state), or no replacement will be done
+  def replace_mappings
+    regions_file = params[:regions]
+    csv = validate_mappings_csv(regions_file)
+    unless csv
+      redirect_to @regional_partner
+      return
+    end
+
+    mappings = []
+    errors = []
+    parse_mappings_from_csv(csv, mappings, errors)
```

Comment:
Instead of passing in output params like this (which reminds me of C++ & C#), you can have the function returns a hash or an array ([example](https://stackoverflow.com/a/28725673)). It is more Rubyist. 
These 3 lines will become 1 line: `mappings, errors = parse_mappings_from_csv(csv)`